hip_data_tools/apache/kafka.py

```diff
@@ -0,0 +1,599 @@
+    """
+    Creates a Kafka consumer
+    Args:
+        bootstrap_servers:  (str): String list of bootstrap servers and their
+            ports e.g. 'localhost:9092' or '[broker1:9092, broker2:9092]'
+        group_id (str) : Group Id for the consumer
+        conf:  string representation of a python dictionary which
+        encapsulates the required settings for the consumer
+
+    Returns: Instantiated Kafka Consumer
+
+    """
+    bootstrap_servers = bootstrap_servers if bootstrap_servers is not \
+                                             None else \
+        get_from_env_or_default_with_warning(
+            "KAFKA_BOOTSTRAP_SERVERS", 'localhost:9092')
+
+    # To prevent any clashes with other consumers, if no group id is
+    # provided we default to a random uuid
+    group_id = group_id if group_id is not \
+                           None else \
+        get_from_env_or_default_with_warning(
+            "KAFKA_CONSUMER_GROUP_ID", "random_id_" + str(uuid.uuid4()))
+
+    configs = literal_eval(get_from_env_or_default_with_warning(
+        "KAFKA_CONSUMER_CONF", DEFAULT_CONSUMER_CONF)) if conf is None else conf
+
+    configs.update({'bootstrap.servers': bootstrap_servers})
+    configs.update({'group.id': group_id})
+
+    return Consumer(configs)
+
+
+def create_kafka_poller(kafka_consumer=None,
+                        topic=None,
+                        timeout_interval=None):
+    """
+    Factory method to create a Kafka Poller and take values from environment
+    if they aren't provided
+    Args:
+        kafka_consumer (Consumer): Kafka Consumer
+        topic (str): Name of the topic to subscribe to
+        timeout_interval (int): Timeout interval for kafka connection
+
+    Returns (KafkaPoller): Instantiated KafkaPoller
+
+    """
+
+    kafka_consumer = kafka_consumer if kafka_consumer is not None \
+        else create_consumer()
+
+    topic = topic if topic is not None else os.environ.get("KAFKA_TOPIC")
+
+    timeout_interval = int(timeout_interval) if timeout_interval is not None \
+        else int(os.environ.get("KAFKA_TIMEOUT_INTERVAL"))
+
+    kafka_poller = KafkaPoller(kafka_consumer, topic, timeout_interval)
+
+    return kafka_poller
+
+
+def create_kafka_batch_s3_uploader(root_path=None,
+                                   bucket=None,
+                                   ts_col_nm=None,
+                                   partition_key_nm=None):
+    """
+    Factory method to generate a kafka batch uploader
+    Args:
+        root_path (str): path for data to be exported
+        bucket (str): Name of bucket for export
+        ts_col_nm (str): Name of the timestamp column used to partition the
+            messages
+        partition_key_nm (str): Name of the column in which to deposit the
+            rounded timestamps used to partition the data
+    Returns (KafkaS3BatchExporter): Instantiated Exporter
+
+    """
+    root_path = root_path if root_path is not None else os.environ.get(
```

Comment:
perhaps read the env vars in a `KafkaConfig` class, and pass in that object here?